hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -501,6 +507,115 @@ public void setPathProperties(final Path path,
     }
   }
 
+
+  /**
+   * Wrapper on {@link #getBlobProperty(Path, TracingContext)} with the handling
+   * for the httpStatusCode = 404 on the server response.
+   *
+   * @param blobPath path for which the property information is required
+   * @param tracingContext object of TracingContext required for the tracing of
+   * server calls
+   * @return instance of BlobProperty if the blob is present on the given path.
+   * <code>null</code> if there is no blob on the given path.
+   * @throws AzureBlobFileSystemException exception other than
+   * {@link AbfsRestOperationException} for httpStatusCode = 404 on the server
+   * response.
+   */
+  BlobProperty getBlobPropertyWithNotFoundHandling(Path blobPath,
```

Comment:
Added.

Reason for not handling this in getBlobProperty(Path, TracingContext) is to keep it in sync with the getFileStatus(Path, TracingContext) behaviour. The getFileStatus(Path, TracingContext) throws the exception it receives to the caller. Hence, it is expected that getBlobProperty(Path, TracingContext) also throw all kind of exception to the caller.

Reply:
Yes, there are multiple occurence where it will take 404 for getBlobProperties as that path doesn't exists. This method provides a way wherein, if it exists, blobProperty obj will be returned, if not, it will return null.

Reply:
removed.